R/biokinetics.R

```diff
@@ -26,6 +28,30 @@ biokinetics <- R6::R6Class(
         stop("Model has not been fitted yet. Call 'fit' before calling this function.")
       }
     },
+    get_upper_detection_limit = function(upper_limit) {
+      if (is.null(upper_limit)) {
+        private$upper_limit <- max(private$data$value)
+      } else {
+        max_value <- max(private$data$value)
+        if (max_value > upper_limit) {
+          warning(sprintf("Data contains a value of %s which is greater than the upper detection limit %s",
```

Comment:
Maybe add the these points will be censored, just so the user knows they are still going to be included

Reply:
Well, actually at the moment we rely on the user specifying exactly which points are censored, via a column in the dataset (see https://seroanalytics.org/epikinetics/articles/data.html#input-data).

But perhaps we should just remove this column and treat any point >= upper limit as censored (similar <= lower limit)?